From a6a6cdb18f6468dabbdcfb0fabb972e9e929056c Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Wed, 3 Jun 2026 21:40:31 +0800
Subject: [PATCH 2/4] VibrateUtils: Switch to standard EFFECT_DOUBLE_CLICK

E AndroidRuntime: java.lang.NullPointerException: null cannot be cast to non-null type com.oplus.os.LinearmotorVibrator
---
 .../gallery/addon/os/VibratorUtils.smali      | 111 +++++++-----------
 1 file changed, 45 insertions(+), 66 deletions(-)

diff --git a/smali_classes10/com/oplus/gallery/addon/os/VibratorUtils.smali b/smali_classes10/com/oplus/gallery/addon/os/VibratorUtils.smali
index 459ee6eec..6d9b610e8 100644
--- a/smali_classes10/com/oplus/gallery/addon/os/VibratorUtils.smali
+++ b/smali_classes10/com/oplus/gallery/addon/os/VibratorUtils.smali
@@ -449,83 +449,25 @@
     .annotation runtime Lkotlin/jvm/JvmStatic;
     .end annotation
 
-    sget-object v0, Lcom/oplus/gallery/addon/os/VibratorUtils;->d:Lcom/oplus/aiunit/vision/kkd;
-
-    if-nez v0, :cond_0
-
-    new-instance v0, Lcom/oplus/aiunit/vision/kkd;
-
-    const-string v1, "context"
-
-    invoke-static {p0, v1}, Lkotlin/jvm/internal/Intrinsics;->checkNotNullParameter(Ljava/lang/Object;Ljava/lang/String;)V
-
-    invoke-direct {v0}, Ljava/lang/Object;-><init>()V
-
-    invoke-virtual {v0, p0}, Lcom/oplus/aiunit/vision/kkd;->a(Landroid/content/Context;)V
-
-    sput-object v0, Lcom/oplus/gallery/addon/os/VibratorUtils;->d:Lcom/oplus/aiunit/vision/kkd;
-
-    :cond_0
-    sget-object p0, Lcom/oplus/gallery/addon/os/VibratorUtils;->d:Lcom/oplus/aiunit/vision/kkd;
-
-    if-eqz p0, :cond_6
-
-    sget-object v0, Lcom/oplus/gallery/addon/os/VibratorUtils$b;->a:[I
+    const-string v0, "context"
+    invoke-static {p0, v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNullParameter(Ljava/lang/Object;Ljava/lang/String;)V
+    const-string v0, "type"
+    invoke-static {p1, v0}, Lkotlin/jvm/internal/Intrinsics;->checkNotNullParameter(Ljava/lang/Object;Ljava/lang/String;)V
 
-    invoke-virtual {p1}, Ljava/lang/Enum;->ordinal()I
+    const-string v0, "vibrator"
+    invoke-virtual {p0, v0}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
+    move-result-object p0
+    check-cast p0, Landroid/os/Vibrator;
 
+    if-eqz p0, :cond_0
+    invoke-virtual {p0}, Landroid/os/Vibrator;->hasVibrator()Z
     move-result p1
+    if-eqz p1, :cond_0
+    const/4 p1, 0x4
+    invoke-static {p1}, Landroid/os/VibrationEffect;->createPredefined(I)Landroid/os/VibrationEffect;
+    move-result-object p1
+    invoke-virtual {p0, p1}, Landroid/os/Vibrator;->vibrate(Landroid/os/VibrationEffect;)V
 
-    aget p1, v0, p1
-
-    const/4 v0, 0x1
-
-    if-eq p1, v0, :cond_4
-
-    const/4 v1, 0x2
-
-    if-eq p1, v1, :cond_3
-
-    const/4 v1, 0x3
-
-    if-eq p1, v1, :cond_5
-
-    const/4 v0, 0x4
-
-    if-eq p1, v0, :cond_2
-
-    const/4 v0, 0x5
-
-    if-ne p1, v0, :cond_1
-
-    const/4 v0, 0x0
-
-    goto :goto_0
-
-    :cond_1
-    new-instance p0, Lkotlin/NoWhenBranchMatchedException;
-
-    invoke-direct {p0}, Lkotlin/NoWhenBranchMatchedException;-><init>()V
-
-    throw p0
-
-    :cond_2
-    const/16 v0, 0x32
-
-    goto :goto_0
-
-    :cond_3
-    move v0, v1
-
-    goto :goto_0
-
-    :cond_4
-    const/16 v0, 0x44
-
-    :cond_5
-    :goto_0
-    invoke-virtual {p0, v0}, Lcom/oplus/aiunit/vision/kkd;->b(I)V
-
-    :cond_6
+    :cond_0
     return-void
 .end method
-- 
2.39.5

